### PR TITLE
check before trying to extend objects

### PR DESCRIPTION
**Pull request by rleemorlang**
Proposed fix for issue described in AndreasMadsen/stack-chain#9

### CODE REVIEW

File: test/simple/non-extensible-errors.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  function performTest() {
+    try {
+      throw error;
+    } catch (e) {
+      e.stack;
+    }
+  }
+
+  t.doesNotThrow(performTest);
```
Comment (AndreasMadsen) [inline test/simple/non-extensible-errors.js]: Nit. Just do 
```
t.doesNotThrow(function () {
  error.stack;
});
```

**Comment by AndreasMadsen** [comment]
Could you add a test for correctness of the `.stack` property. 

```javascript
test('stack is correct on non extensible error object', function (t) {
  var error = new Error("don't extend me");
  Object.preventExtensions(error);

  chain.format.replace(function () {
    return 'good';
  });

  try {
    t.equal(error.stack, 'good');
  } catch (e) { t.ifError(e); }

  chain.format.restore();

  t.end();
});
```

**Comment by rleemorlang** [comment]
Thanks. Will revise.

**Comment by AndreasMadsen** [comment]
Lovely. Thanks

**Pull request merged**